src/game/AI/ScriptDevAI/scripts/kalimdor/darkshore.cpp

```diff
@@ -225,126 +225,79 @@ bool EffectDummyCreature_awaken_kerlonian(Unit* pCaster, uint32 uiSpellId, Spell
-            case 43:
-                DoScriptText(EMOTE_REM_END, m_creature, pPlayer);
-                pPlayer->RewardPlayerAndGroupAtEventExplored(QUEST_ABSENT_MINDED_PT2, m_creature);
-                break;
+            case 0: DoBroadcastText(SAY_REM_AGGRO_1, m_creature); break;
+            case 1: DoBroadcastText(SAY_REM_AGGRO_2, m_creature, who); break;
+            case 2: DoBroadcastText(SAY_REM_AGGRO_3, m_creature, who); break;
+            case 3: DoBroadcastText(SAY_REM_AGGRO_4, m_creature, who); break;
+            case 4: DoBroadcastText(SAY_REM_AGGRO_5, m_creature, who); break;
+            case 5: DoBroadcastText(SAY_REM_AGGRO_6, m_creature, who); break;
         }
     }
 
-    void Reset() override { }
-
-    void Aggro(Unit* pWho) override
+    void ReceiveAIEvent(AIEventType eventType, Unit* sender, Unit* /*invoker*/, uint32 /*miscValue*/) override
     {
-        if (urand(0, 1))
-            DoScriptText(SAY_REM_AGGRO, m_creature, pWho);
+        if (eventType == AI_EVENT_CUSTOM_EVENTAI_A)
+            if(m_creature->IsAlive())
+                sender->AI()->AttackStart(m_creature);        
     }
 
-    void JustSummoned(Creature* /*pSummoned*/) override
+    void WaypointReached(uint32 uiPointId) override
     {
-        // unsure if it should be any
-        // pSummoned->AI()->AttackStart(m_creature);
+        switch (uiPointId)
+        {
+            case 38:
+                if (Player* player = GetPlayerForEscort())
+                    player->RewardPlayerAndGroupAtEventExplored(QUEST_ABSENT_MINDED_PT2, m_creature);
+                break;
+            case 39:
+                m_creature->GetMotionMaster()->Clear(false, true);
+                m_creature->GetMotionMaster()->MoveIdle();
+                End();
+                break;
+        }
     }
 };
 
-UnitAI* GetAI_npc_prospector_remtravel(Creature* pCreature)
-{
-    return new npc_prospector_remtravelAI(pCreature);
-}
-
-bool QuestAccept_npc_prospector_remtravel(Player* pPlayer, Creature* pCreature, const Quest* pQuest)
+bool QuestAccept_npc_prospector_remtravel(Player* player, Creature* creature, const Quest* quest)
 {
-    if (pQuest->GetQuestId() == QUEST_ABSENT_MINDED_PT2)
+    if (quest->GetQuestId() == QUEST_ABSENT_MINDED_PT2)
     {
-        pCreature->SetFactionTemporary(FACTION_ESCORT_A_PASSIVE, TEMPFACTION_RESTORE_RESPAWN | TEMPFACTION_TOGGLE_IMMUNE_TO_NPC);
-
-        if (npc_prospector_remtravelAI* pEscortAI = dynamic_cast<npc_prospector_remtravelAI*>(pCreature->AI()))
-            pEscortAI->Start(false, pPlayer, pQuest, true);
+        if (npc_prospector_remtravelAI* pEscortAI = dynamic_cast<npc_prospector_remtravelAI*>(creature->AI()))
+        {
+            creature->SetFactionTemporary(FACTION_ESCORT_A_PASSIVE, TEMPFACTION_RESTORE_RESPAWN);
+            pEscortAI->Start(false, player, quest, false, false, REMTRAVEL_ESCORT_PATH);
+        }
     }
-
     return true;
 }
 
+UnitAI* GetAI_npc_prospector_remtravel(Creature* creature)
```

Comment:
Redundant function like in different pr and ur also not assigning script to script name below